frontend/public/components/utils/firehose.jsx

```diff
@@ -155,40 +189,19 @@ export const Firehose = connect(
       );
     }
 
-    componentWillUnmount () {
-      const { stopK8sWatch } = this.props;
-
-      this.firehoses.forEach(({id}) => stopK8sWatch(id));
+    clear() {
+      this.firehoses.forEach(({id}) => this.props.stopK8sWatch(id));
       this.firehoses = [];
     }
 
-    shouldComponentUpdate(nextProps, nextState, nextContext) {
-      const {resources: currentResources, forceUpdate = false} = this.props;
-
-      const { resources, expand, inFlight } = nextProps;
-
-      if (_.intersectionWith(resources, currentResources, _.isEqual).length === resources.length && inFlight === this.props.inFlight) {
-        if (_.get(nextContext, 'router.route.location.pathname') !== _.get(this.context, 'router.route.location.pathname')) {
-          return true;
-        }
-        if (expand !== this.props.expand) {
-          return true;
-        }
-        return forceUpdate;
-      }
-      this.componentWillUnmount();
-      this.componentWillMount(nextProps);
-      return true;
-    }
-
     render () {
       const reduxes = this.firehoses.map(({id, prop, isList, filters, optional}) => ({reduxID: id, prop, isList, filters, optional}));
       const children = inject(this.props.children, _.omit(this.props, [
         'children',
         'className',
       ]));
 
-      return this.props.inFlight ? null : <ConnectToState reduxes={reduxes}> {children} </ConnectToState>;
+      return this.props.loaded ? <ConnectToState reduxes={reduxes}> {children} </ConnectToState> : null;
```

Comment:
At some point, we should consider adding a loading animation while discovery is in flight. The page is white while it's happening, and it takes a while on a slow network.

(Not necessarily for this PR. Just an observation.)

Reply:
I agree, the console could use some nice animations.